scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3130,15 +3477,64 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       )
     }
 
-    Defn.Class(mods, className, typeParams, ctor, templateOpt(OwnedByClass))
+    val tmpl = templateOpt(OwnedByClass)
+    Defn.Class(mods, className, typeParams, ctor, tmpl)
+  }
+
+  def enumDef(mods: List[Mod]): Defn.Enum = atPos(mods, auto) {
+    // EnumDef           ::=  id ClassConstr InheritClauses EnumBody
+    accept[KwEnum]
+
+    val enumName = typeName()
+    val culprit = s"enum $enumName"
+    onlyAllowedMods[Mod.Private, Mod.Protected](mods, culprit)
+
+    val typeParams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = true)
+    val ctor = primaryCtor(OwnedByEnum)
+    val tmpl = templateOpt(OwnedByEnum)
+    val cases = tmpl.stats.collect { case _: Defn.Case | _: Defn.RepeatedCase => true }
```

Comment:
```suggestion
    val containsCase = tmpl.stats.find { 
      case _: Defn.Case | _: Defn.RepeatedCase => true
      case _ => false
     }
```
No need to go through the entire collection.

Reply:
good catch :+1: